MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs

```diff
@@ -18,22 +19,45 @@ public MosaicGateway(MosaicContext mosaicContext)
             _mosaicContext = mosaicContext;
         }
 
-        public List<ResidentInformation> GetAllResidents()
+        public List<ResidentInformation> GetAllResidents(string firstname = null, string lastname = null, string postcode = null)
         {
-            var persons = _mosaicContext.Persons.ToList();
+            var persons = _mosaicContext.Persons
+                .Where(p => string.IsNullOrEmpty(firstname) ? true : p.FirstName.ToLower().Equals(firstname.ToLower()))
+                .Where(p => string.IsNullOrEmpty(lastname) ? true : p.LastName.ToLower().Equals(lastname.ToLower()))
+                .ToList();
 
             var personDomain = persons.ToDomain();
 
-            foreach (var person in personDomain)
+            var personsPostcode = _mosaicContext.Addresses
+                .Include(p => p.Person)
+                .Where(a => a.PostCode.Equals(postcode))
+                .ToList();
+
+            var ppc = personsPostcode
```

Comment:
It would be good to have more descriptive variables names here, I'm not sure what `ppc` is

Reply:
its the acronym of the right hand value